app/grandchallenge/documentation/models.py

```diff
@@ -0,0 +1,185 @@
+from django.core.exceptions import ObjectDoesNotExist
+from django.db import models
+from django.db.models import Max
+from django.utils.translation import gettext_lazy as _
+from django_extensions.db.fields import AutoSlugField
+from simple_history.models import HistoricalRecords
+
+from grandchallenge.core.templatetags.bleach import clean
+from grandchallenge.core.utils.query import index
+from grandchallenge.subdomains.utils import reverse
+
+
+class LevelChoices(models.IntegerChoices):
+    """Documentation page level options"""
+
+    PRIMARY = 1, _("Top level")
+    SECONDARY = 2, _("Second level")
+    TERTIARY = 3, _("Third level")
+
+
+class DocPageLevel:
+    """Documentation page level."""
+
+    DocPageLevelChoices = LevelChoices
+
+
+class DocPage(models.Model):
+
+    Level = DocPageLevel.DocPageLevelChoices
+
+    UP = "UP"
+    DOWN = "DOWN"
+    FIRST = "FIRST"
+    LAST = "LAST"
+
+    created = models.DateTimeField(auto_now_add=True)
+    modified = models.DateTimeField(auto_now=True)
+
+    title = models.CharField(max_length=1024)
+    slug = AutoSlugField(populate_from="title", max_length=1024)
+    display_title = models.CharField(
+        max_length=255,
+        default="",
+        blank=True,
+        help_text=(
+            "On pages and in menu items, use this text. Spaces and special "
+            "chars allowed here. Optional field. If emtpy, title is used"
+        ),
+    )
+
+    content = models.TextField()
+
+    level = models.PositiveSmallIntegerField(
+        default=Level.PRIMARY,
+        choices=Level.choices,
+        help_text=(
+            "As which level should this page be displayed in the sidebar?"
+        ),
+    )
+
+    order = models.IntegerField(
+        editable=False,
+        default=1,
+        help_text="Determines order in which pages appear in side menu",
+    )
+
+    history = HistoricalRecords(excluded_fields=["level", "order"])
+
+    def __str__(self):
+        return self.title
+
+    def save(self, *args, **kwargs):
+        # when saving for the first time only, put this page last in order
+        if not self.id:
+            # get max value of order for current pages.
+            try:
+                self.order = (
+                    DocPage.objects.aggregate(Max("order"))["order__max"] + 1
+                )
+            except ObjectDoesNotExist:
+                # Use the default
+                pass
+            except TypeError:
+                # Use the default
+                pass

```

Comment:
You can group exceptions if you're going to take the same action:

```suggestion
            except (ObjectDoesNotExist, TypeError):
                # Use the default
                pass

```